src/splits.cpp

```diff
@@ -637,9 +638,58 @@ RawMatrix pack_splits_logical_vec(LogicalVector x) {
     
     unsigned char byte = 0;
     for (int k = 0; k < limit; ++k) {
-      if (p[k] == TRUE) byte |= (1u << k);
     }
     out(0, b) = byte;
   }
   return out;
 }
+
+// Fast count of unique splits in a phylogenetic tree
+// Optimized replacement for collapse.singles(x)[["Nnode"]] - 1L - TreeIsRooted(x)
+// [[Rcpp::export]]
+int cpp_count_splits(const Rcpp::IntegerMatrix& edge, int nTip) {
+  if (edge.ncol() != 2) {
+    Rcpp::stop("Edge matrix must contain two columns");
+  }
+  
+  const int n_edge = edge.nrow();
+  if (n_edge == 0 || nTip < 4) {
+    return 0;
+  }
+  
+  // Get all unique parent nodes - these represent internal nodes
+  std::set<int> internal_nodes;
```

Comment:
Any gain to reserving some memory here? We'll often be dealing with binary trees, so expect n_internal to be ~nTip.